### PR TITLE
Return a user-visible error if the project already exists

### PR DESCRIPTION
**Pull request by JAORMX**
# Summary

This indicates to the user if the project they're trying to provision already exists; which may
prompt them to contact their minder administrators.

## Change Type

***Mark the type of change your PR introduces:***

- [ ] Bug fix (resolves an issue without affecting existing features)
- [x] Feature (adds new functionality without breaking changes)
- [ ] Breaking change (may impact existing functionalities or require documentation updates)
- [ ] Documentation (updates or additions to documentation)
- [ ] Refactoring or test improvements (no bug fixes or new functionality)

# Testing

***Outline how the changes were tested, including steps to reproduce and any relevant configurations. 
Attach screenshots if helpful.***

# Review Checklist:

- [x] Reviewed my own code for quality and clarity.
- [ ] Added comments to complex or tricky code sections.
- [ ] Updated any affected documentation.
- [ ] Included tests that validate the fix or feature.
- [ ] Checked that related changes are merged.

### CODE REVIEW

File: internal/projects/create.go [hunk 1]
```diff
@@ -70,7 +76,11 @@ func ProvisionSelfEnrolledProject(
 		Metadata: jsonmeta,
 	})
 	if err != nil {
-		return nil, status.Errorf(codes.Internal, "failed to create default project: %v", err)
+		// Check if `project_name_lower_idx` unique constraint was violated
```
Comment (jhrozek) [inline internal/projects/create.go:79]: Can you use `ErrIsUniqueViolation` instead?

Comment (JAORMX) [reply internal/projects/create.go:79]: Do you want me to rename the error to that? or do you want me to use `ErrIsUniqueViolation` to detect the error? I don't find that error being provided by the SQL package.

Comment (jhrozek) [reply internal/projects/create.go:79]: Sorry I was unclear, I wanted to ask if you think that using https://github.com/stacklok/minder/blob/main/internal/db/errors.go#L24 would be cleaner than checking for the string in Error

Comment (JAORMX) [reply internal/projects/create.go:79]: Oooh! I forgot that existed

**Review by jhrozek — APPROVED** [review]

**Pull request merged**